libs/community/langchain_community/utilities/pubmed_extract_fulltext.py

```diff
@@ -0,0 +1,603 @@
+from __future__ import absolute_import, print_function, unicode_literals
+
+import logging
+import os
+import os.path
+import shutil
+import xml.etree.ElementTree as ET
+from builtins import str
+from typing import Any, ClassVar, Dict
+
+import boto3
+import requests
+from botocore import UNSIGNED
+from botocore.config import Config
+from bs4 import BeautifulSoup
+from indra.literature import pubmed_client
+from indra.util import UnicodeXMLTreeBuilder as UTB
+from langchain_core.pydantic_v1 import BaseModel, root_validator
+from lxml import etree
+from lxml.etree import QName
+from pydantic import BaseModel as PydanticBaseModel
+from pydantic import ConfigDict
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class PmcIDSearchWrapper(BaseModel):
+    """
+    Wrapper around PubMed API.
+
+    This wrapper will use the PMC API to conduct pmc_id searches and fetch
+    the full text of paper. By default, it will return the full text of paper with the pmc_id
+
+    Parameters:
+
+        pmc_id: the unique id of pubmed central paper
+    """
+
+    # model_config = ConfigDict(ignored_types=(IgnoredType,))
+    parse: Any  #: :meta private:
+
+    # Default values for the parameters
+
+    pmc_url = "https://www.ncbi.nlm.nih.gov/pmc/oai/oai.cgi"
+    pmid_convert_url = "https://www.ncbi.nlm.nih.gov/pmc/utils/idconv/v1.0/"
+
+    # Paths to resource files
+    pmids_fulltext_path = os.path.join(os.path.dirname(__file__), "pmids_fulltext.txt")
```

Comment:
What are these resource files? Are they ever used by the integration?

Reply:
Hi Eugene, Thanks for quick review and suggestions. I will update the PR. 
These resource can useful to retrieve full-text by pmid. Convert pmid to pmcid, then retrieve full-text. I will add a function to achieve it. Thanks.